client/src/app/shared/directives/only-number.directive.ts

```diff
@@ -0,0 +1,45 @@
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+
+@Directive({
+    selector: '[osOnlyNumber]'
+})
+export class OnlyNumberDirective {
+    @Input()
+    public osOnlyNumber = true;
+
+    /**
+     * Regex to validate only numbers
+     * ^: starts with
+     * $: ends
+     */
+    private regExp = new RegExp('^[0-9]*$');
```

Comment:
^([1-9][0-9]*|0)?$

maybe ddisallow a leading zeros. Whats about a minus? Maybe add a docstring what exactly this directive allows and what now.